hbase-client/src/main/java/org/apache/hadoop/hbase/client/Admin.java

```diff
@@ -1633,7 +1633,25 @@ void restoreSnapshot(String snapshotName, boolean takeFailSafeSnapshot, boolean
    */
   default void cloneSnapshot(String snapshotName, TableName tableName)
       throws IOException, TableExistsException, RestoreSnapshotException {
-    cloneSnapshot(snapshotName, tableName, false);
+    cloneSnapshot(snapshotName, tableName, false, null);
+  }
+
+  /**
+   * Create a new table by cloning the snapshot content.
+   * @param snapshotName name of the snapshot to be cloned
+   * @param tableName name of the table where the snapshot will be restored
+   * @param restoreAcl <code>true</code> to clone acl into newly created table
+   * @param customSFT specify the StroreFileTracker used for the table
+   * @throws IOException if a remote or network exception occurs
+   * @throws TableExistsException if table to be created already exists
+   * @throws RestoreSnapshotException if snapshot failed to be cloned
+   * @throws IllegalArgumentException if the specified table has not a valid name
+   */
+  default void cloneSnapshot(String snapshotName, TableName tableName, boolean restoreAcl,
+    String customSFT)
```

Comment:
Admin is public API, so this is a commitment that using a String for SFT is the path forward. Do we want a String vs. an enum? I'm not positive if our rules say that we can mark just this one method as private/unstable until we are sure SFT is ready for all users.

Reply:
Ok, Admin is IA.Public which requires it to be IS.Stable. Rereading the hbase book, I don't think there's anything stopping us from claiming this one method is IA.LimitedPrivate/IS.Evolving. Maybe someone else can verify my thinking.

Reply:
I would prefer to keep it String. The current logic do handle both class names provided as strings and enums. While the enum simplifies SFT handling I see no reason to limit pluggability at this point.

What I have found matches your description so I'm adding IS.Evolving for now.

Reply:
For IA.Public interface, all methods in it should be considered IS.Stable. The IS annotation is only useful when the class is IA.LimitedPrivate. For IA.Private, everything is IS.Unstable. That's my understanding.

And I think a String is OK here, we could do some checks at server side when we actually convert it to a SFT instance, and report back to client if there is something wrong.